content/doc/pipeline/declarative/Using-multiple-steps,-stages-and-notifications.adoc

```diff
@@ -0,0 +1,66 @@
+You have probably landed here because you are in a hurry, and want to
+eyeball a real example. Here is one that builds the pipeline config
+plugin itself (it builds itself, in itself, also known as 'eating your
+own dogfood'):
+
+[source,groovy]
+----
+pipeline {
+    tools {
+        maven "Maven 3.3.9"
+        jdk "Oracle JDK 8u40"
+    }
+
+    agent label:"any-executor"
+
+    stages {
+        stage("build") {
+            steps {
+                sh 'echo "path: ${PATH}"'
+                sh 'echo "M2_HOME: ${M2_HOME}"'
+                sh 'mvn clean install -Dmaven.test.failure.ignore=true'
+            }
+        }
+    }
+
+    post {
+        always {
+            archive "target/**/*"
+            junit 'target/surefire-reports/*.xml'
+        }
+        success {
+            mail to:"me@example.com", subject:"SUCCESS: ${currentBuild.fullDisplayName}", body: "Yay, we passed."
+        }
+        failure {
+            mail to:"me@example.com", subject:"FAILURE: ${currentBuild.fullDisplayName}", body: "Boo, we failed."
+        }
+        unstable {
+            mail to:"me@example.com", subject:"UNSTABLE: ${currentBuild.fullDisplayName}", body: "Huh, we're unstable."
+        }
+        changed {
+            mail to:"me@example.com", subject:"CHANGED: ${currentBuild.fullDisplayName}", body: "Wow, our status changed!"
+        }
+    }
+}
+----
+
+Let's tear this down a bit.
```

Comment:
There is a code footnote marking in adoc: 
```
     tools { // <1>
```

```
<1> There is a `tool` installation section. Jenkins has the concept of
 ```

See this for example: 
https://github.com/jenkins-infra/jenkins.io/blame/master/content/blog/2016/2016-08-29-sauce-pipeline.adoc#L72